### PR TITLE
Race condition fix

### PR DESCRIPTION
**Pull request by Smitters**
## Overview
If a user rotates the device while creating an entity, the app will crash after the entity is saved. This appears to only affect entities with split views, as it is not reproducible with the current implementation of invoices, for example.

Steps:
1. Tap "New".
2. Rotate the device.
3. Enter valid values into all mandatory fields, then tap "Save".

See [TICKET](https://freshbooks.atlassian.net/browse/HEL-12)

#### Notes for Devs
Starting with v4.0 and Realm v0.99, RBQFetchedResultsController works automatically with Realm's fine-grained notification support. It is recommended to use Realm's API directly instead of RBQFetchedResultsController **unless you need support for sections.**

We could add to backlog task to get rid of RBQFetchedResultsController and use Realm's API directly since we don't use sections here.

[Branch](https://github.com/freshbooks/helios/tree/HEL-12-save-entity-on-rotating) with this fix for testing.

[Build](https://dashboard.buddybuild.com/apps/5a0b5f6c4f46da0001afa890/build/5aa27cd3a54226000170f8d3) with this fix on buddybuild. 

#### Checklist:
- [ ] I have added/updated unit tests to cover my business logic changes where possible.
- [ ] I have added/updated snapshot tests for UI changes where possible.
- [x] I have linted my changes.
- [ ] I have checked the code coverage of my changed code.

### CODE REVIEW

File: RBQFetchedResultsController/Source/RBQFetchedResultsController.m
Comment (ccheungfb) [inline RBQFetchedResultsController/Source/RBQFetchedResultsController.m:820]: This method is way too long holy smokes.

So the problem is that the executing thread are doing things but it needs to be be blocked by main?  Looks like you found the solution lower in the code 👍

**Review by ccheungfb — APPROVED** [review]
Good job at finding the cause of the problem.  Can you please check if there are other areas that require the use of semaphores?

**Review by maddiepage — APPROVED** [review]
This seems fine to me. Can we add some additional commenting around where you have added code? Just something to identify why it was added/who added it.

I've also asked @stranfb to take a look, once he approves we can merge this.

**Review by stranfb — APPROVED** [review]
I would agree with @maddiepage having some comments would help provide clarity on this fix.

**Pull request merged**
